rqt_image_view/src/rqt_image_view/image_view.cpp

```diff
@@ -280,6 +292,49 @@ void ImageView::saveImage()
   img.save(file_name);
 }
 
+void ImageView::onMousePublish(bool checked)
+{
+  if(checked)
+  {
+    std::string topicName;
+    if(pub_topic_custom_)
+    {
+      topicName = ui_.publish_click_location_topic_line_edit->text().toStdString();
+    } else {
+      if(!subscriber_.getTopic().empty())
+      {
+        topicName = subscriber_.getTopic()+"_mouse_left";
+      } else {
+        topicName = "mouse_left";
+      }
+      ui_.publish_click_location_topic_line_edit->setText(QString::fromStdString(topicName));
+    }
+    pub_mouse_left_ = getNodeHandle().advertise<geometry_msgs::Point>(topicName, 1000);
+  } else {
+    pub_mouse_left_.shutdown();
+  }
+}
+
+void ImageView::onMouseLeft(int x, int y)
+{
+  if(ui_.publish_click_location_check_box->isChecked() && !ui_.image_frame->getImage().isNull())
+  {
+    geometry_msgs::Point clickLocation;
+    // Publish click location in pixel coordinates
+    clickLocation.x = round((double)x/(double)ui_.image_frame->width()*(double)ui_.image_frame->getImage().width());
+    clickLocation.y = round((double)y/(double)ui_.image_frame->height()*(double)ui_.image_frame->getImage().height());
+    clickLocation.z = 0;
+    pub_mouse_left_.publish(clickLocation);
+  }
+}
+
+void ImageView::onPubTopicChanged()
+{
+  pub_topic_custom_ = !(ui_.publish_click_location_topic_line_edit->text().isEmpty());
+  pub_mouse_left_.shutdown();
```

Comment:
Why is this necessary? The function `onMousePublish` either create a new publisher or shuts down the previous one.